### PR TITLE
[Merged by Bors] - restart syncing pod in sync auto

### PR DESCRIPTION
**Pull request by almogdepaz**

### CODE REVIEW

File: tests/sync/test_sync.py [hunk 1]
```diff
@@ -66,30 +68,40 @@ def test_sync_gradually_add_nodes(init_session, setup_bootstrap, save_log_on_exi
...
 
     start = time.time()
     sleep = 30 # seconds
     num_iter = 20 # total of 5 minutes
     for i in range(num_iter):
         done = 0
         for j in range(0, num_clients):
-            podName = clients[j].pods[0]['name']
-            if not check_pod(podName): # not all done
+            podName = clients[j]
```
Comment (AmitShaul) [inline tests/sync/test_sync.py:106]: once you pull from develop you'll see the new functionality I added.
please use the functions I created there

File: tests/sync/test_sync.py [hunk 2]
```diff
@@ -66,30 +68,40 @@ def test_sync_gradually_add_nodes(init_session, setup_bootstrap, save_log_on_exi
...
-    inf = new_client_in_namespace(testconfig['namespace'], setup_bootstrap, cspec, 10)
+    inf = add_multi_clients(setup_bootstrap.deployment_id, cspec, 10)
 
     del cspec.args['remote-data']
     cspec.args['data-folder'] = ""
 
     num_clients = 4
     clients = [None] * num_clients
-    clients[0] = new_client_in_namespace(testconfig['namespace'], setup_bootstrap, cspec, 1)
+    clients[0] = add_multi_clients(setup_bootstrap.deployment_id, cspec2, 1,'clientv2')[0]
```
Comment (AmitShaul) [inline tests/sync/test_sync.py]: setup_bootstrap.deployment_id = init_session

File: tests/sync/test_sync.py [hunk 3]
```diff
@@ -66,30 +68,40 @@ def test_sync_gradually_add_nodes(init_session, setup_bootstrap, save_log_on_exi
...
     start = time.time()
     sleep = 30 # seconds
     num_iter = 20 # total of 5 minutes
     for i in range(num_iter):
         done = 0
         for j in range(0, num_clients):
-            podName = clients[j].pods[0]['name']
-            if not check_pod(podName): # not all done
+            podName = clients[j]
+            if not check_pod(podName,{"M": "sync done"}): # not all done
```
Comment (AmitShaul) [inline tests/sync/test_sync.py]: once you pull from develop you'll see the new functionality I added.
please use the functions I created there

File: tests/sync/test_sync.py [hunk 4]
```diff
@@ -107,7 +119,8 @@ def test_sync_gradually_add_nodes(init_session, setup_bootstrap, save_log_on_exi
 
     end = time.time()
 
-    delete_deployment(inf.deployment_name, testconfig['namespace'])
+    check_pod(clients[0],{"M": "persistent data found"})
```
Comment (AmitShaul) [inline tests/sync/test_sync.py]: once you pull from develop you'll see the new functionality I added.
please use the functions I created there

File: tests/sync/test_sync.py [hunk 5]
```diff
@@ -107,7 +119,8 @@ def test_sync_gradually_add_nodes(init_session, setup_bootstrap, save_log_on_exi
...
 
     print("it took " + str(end - start) + " to sync all nodes with " + cspec.args['expected-layers'] + "layers")
+    time.sleep(3*60)
```
Comment (AmitShaul) [inline tests/sync/test_sync.py]: can you please add a print that you're sleeping here, also why sleeping here?

File: tests/sync/test_sync.py (hunk 3 above)
Comment (almogdepaz) [reply tests/sync/test_sync.py]: im leaving the check pod its convenient for this use

File: tests/sync/test_sync.py (hunk 5 above)
Comment (almogdepaz) [reply tests/sync/test_sync.py]: legacy code, deleted

**Review by AmitShaul — APPROVED** [review]

**Comment by almogdepaz** [comment]
bors retry

**Pull request closed without merging**